hips/tiles/tests/test_tile.py

```diff
@@ -4,45 +4,103 @@
+        # the data is different (for unknown reasons).
+        # TODO: Figure out what's wrong here and fix!
+        # print(tile1.data.sum())
+        # print(tile2.data.sum())
+        # print((tile1 == tile2).all())
+        # assert tile1 == tile2
+
+    def assert_png(self, tile1, tile2):
+        assert tile1.data.shape == (512, 512, 4)
+        assert_equal(tile1.data[254][5:7], [[15, 15, 15, 255], [14, 14, 14, 255]])
+        assert tile1 == tile2
+
     @remote_data
     def test_fetch_read_write_fits(self, tmpdir):
-        meta = HipsTileMeta(order=6, ipix=30889, file_format='fits')
-        url = 'http://alasky.unistra.fr/2MASS/H/Norder6/Dir30000/Npix30889.fits'
+        meta = HipsTileMeta(order=3, ipix=463, file_format='fits')
+        url = 'http://alasky.unistra.fr/DSS/DSS2Merged/Norder3/Dir0/Npix463.fits'
         tile = HipsTile.fetch(meta, url)
 
-        assert tile.data.shape == (512, 512)
-        assert_equal(tile.data[510][5:7], [1, 0])
-
-        filename = str(tmpdir / 'Npix30889.fits')
+        filename = str(tmpdir / 'Npix463.fits')
         tile.write(filename)
-        tile2 = HipsTile.read(meta, filename=filename)
+        tile2 = HipsTile.read(meta, full_path=filename)
 
-        assert tile == tile2
+        self.assert_fits(tile, tile2)
 
     @remote_data
     def test_fetch_read_write_jpg(self, tmpdir):
-        meta = HipsTileMeta(order=6, ipix=30889, file_format='jpg')
-        url = 'http://alasky.unistra.fr/2MASS/H/Norder6/Dir30000/Npix30889.jpg'
+        meta = HipsTileMeta(order=3, ipix=463, file_format='jpg')
+        url = 'http://alasky.unistra.fr/DSS/DSS2Merged/Norder3/Dir0/Npix463.jpg'
         tile = HipsTile.fetch(meta, url)
 
-        assert tile.data.shape == (512, 512, 3)
-        assert_equal(tile.data[510][5:7], [[0, 0, 0], [1, 1, 1]])
+        filename = str(tmpdir / 'Npix463.jpg')
+        tile.write(filename)
+        tile2 = HipsTile.read(meta, full_path=filename)
+
+        self.assert_jpg(tile, tile2)
 
-        filename = str(tmpdir / 'Npix30889.jpg')
+    @remote_data
+    def test_fetch_read_write_png(self, tmpdir):
+        meta = HipsTileMeta(order=6, ipix=6112, file_format='png')
+        url = 'http://alasky.unistra.fr/2MASS6X/2MASS6X_H/Norder6/Dir0/Npix6112.png'
+        tile = HipsTile.fetch(meta, url)
+
+        filename = str(tmpdir / 'Npix6112.png')
         tile.write(filename)
-        tile2 = HipsTile.read(meta, filename=filename)
+        tile2 = HipsTile.read(meta, full_path=filename)
 
-        # The following assert fails, because on JPEG write / read
-        # the data is different (for unknown reasons).
-        # TODO: Figure out what's wrong here and fix!
-        # print(tile.data.sum())
-        # print(tile2.data.sum())
-        # print((tile == tile2).all())
-        # assert tile == tile2
+        self.assert_png(tile, tile2)
+
+    @requires_hips_extra()
+    def test_read_write_fits(self, tmpdir):
```

Comment:
It would be possible to use `pytest.mark.parametrize` to cover the three cases of PNG, JPEG and FITS with just one common function. @adl1995 - If you think it's a good idea and know how to do it, you can make this change.
But the gain is very small, and there might be other asserts that we want to do specifically for each type, so I'm also OK to leave as-is.